realm/realm-library/src/main/java/io/realm/sync/permissions/PermissionUser.java

```diff
@@ -72,4 +74,18 @@ public String getId() {
     public @Nullable  RealmResults<Role> getRoles() {
         return roles;
     }
+
+    /**
+     * The user's private role. This will be initialized to a role named for the user's
+     * identity that contains this user as its only member.
+     *
+     * @return User private {@link Role}.
+     */
+    public Role getRole() {
+        return role;
+    }
+
+    public void setRole(Role role) {
```

Comment:
Add javadoc. Perhaps something like this:

```
Sets a role for this user that can be accessed later. When the PermissionUser object is created this will be initialized with a role named for the user's identity that contains the user as its only member.

```

Reply:
This method shouldn't exist, the `role` is built-in automatically at the Sync client, you can not set  it since `Role` is immutable as far as permission is concerned

Reply:
Not sure what you mean? `Role` isn't immutable since you can/remove members all the time. If you mean `PermissionUser` then you are correct the name of the user is a key and hence immutable, but apart from that, it is just a standard Object.

There are many things about this property I find problematic. See https://github.com/realm/realm-sync/pull/2068#discussion_r173429225

I agree that exposing `setRole()` is just confusing and having it means that people can accidentally set it to what-ever. In any case, how it works, should be aligned across bindings.

Personally, I think having a `getPrivateRole()` only getter would be much more in line with the intended behaviour.

Reply:
I agree `setRole` should be removed, furthermore, the user should not create `PermissionUser` directly it's more of an internal API, it's less likely that you ever want to create user directly, let alone setting the `PermissionUser` built-in Role.